x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/ResponseXContentUtils.java

```diff
@@ -103,7 +107,11 @@ static Iterator<? extends ToXContent> rowValues(List<ColumnInfoImpl> columns, Li
             assert page.getBlockCount() == columnCount : page.getBlockCount() + " != " + columnCount;
             final PositionToXContent[] toXContents = new PositionToXContent[columnCount];
             for (int column = 0; column < columnCount; column++) {
-                toXContents[column] = PositionToXContent.positionToXContent(columns.get(column), page.getBlock(column), scratch);
+                Block block = page.getBlock(column);
+                if (columns.get(column).type() == DataType.AGGREGATE_METRIC_DOUBLE && block instanceof CompositeBlock compositeBlock) {
```

Comment:
We can cast directly to AggregateMetricDoubleBlock with the changes in Block#read,write.